autor.py

```diff
@@ -0,0 +1,48 @@
+from datetime import date
+import re
+
+class Autor:
+    
+    def __init__(self, nome, email, descricao):
+        self.__set_nome(nome)
+        self.__set_email(email)
+        self.__set_descricao(descricao)
+        self.__data_registro = date.today()
+
+    def __set_nome(self, nome):
+        
+        #Verifica se nome eh nulo, vazio ou em branco
+        if nome in (None, '', ' '):
+            raise Exception("Nome nulo")
+
+        self.__nome = nome
+
+    def __set_email(self, email):
+        
+        regex = '^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$'
+        
+        #verifica se email eh valido
+        if not re.search(regex,email):
+            raise Exception("Email errado")
+
+        self.__email = email
+    
+    def __set_descricao(self, descricao):
+
+        #verifica se descricao esta em branco
+        if not descricao :
+            raise Exception("Descricao Nula")
```

Comment:
Aqui eu fiquei curioso. Veja que na validação do nome, você faz: `if nome in (None, '', ' '):`
Mas na descrição você faz diferente, mesmo a validação sendo a mesma: nulo, em branco ou vazio. Isso quer dizer que se for um nulo para a descrição, len vai dar erro, por que vai tentar calcular o tamanho de nulo, mas a mensagem não é de negócio, mas sim da api do python.

Reply:
Verdade, da maneira que está se o usuário passar apenas um espaço em branco é aceito e como a regra a para e-mail e nome é a mesma, não faz sentido as condições serem diferentes.
Porem não vou ter problema no len porque o if anterior verifica se é nulo e já lança uma exceção, então o len não é executado.